doc-surrealdb_versioned_docs/version-2.x/reference-guide/security-best-practices.mdx

```diff
@@ -0,0 +1,323 @@
+---
+sidebar_position: 4
+sidebar_label: Security Best Practices
+title: Security Best Practices | Reference guides
+description: This guide outlines some key security best practices for using SurrealDB 2.0. While SurrealDB offers powerful and flexible features to support you in meeting your desired security standards, the use that you make of those features will ultimately determine whether or not you meet them.
+---
+import ThemedImage from '@theme/ThemedImage';
+import useBaseUrl from '@docusaurus/useBaseUrl';
+import { SurrealistMini } from "@site/src/components/SurrealistMini";
+
+# Security Best Practices
+
+This guide outlines some key security best practices for using SurrealDB 2.0. While SurrealDB offers powerful and flexible features to support you in meeting your desired security standards, the use that you make of those features will ultimately determine whether or not you meet them. The following is a non-exhaustive list of security best practices that you should take into account when building services and applications with SurrealDB to help you address common security challenges while preventing frequent pitfalls.
+
+## Capabilities
+
+When running a SurrealDB server, you are able to configure the [capabilities](https://surrealdb.com/docs/surrealdb/security/capabilities) that it will have. These capabilities are disabled by default to expose as little attack surface as possible to malicious actors. We strongly discourage running SurrealDB with all capabilities enabled without a good reason for it. Instead, we recommend enabling only the capabilities that are necessary to your service.
+
+Whenever you need to enable a specific capability, we recommend that you do it on a case by case basis instead of generally. For example, if you know that your queries need to be able to parse emails using functions, we recommend you run SurrealDB with the `--allow-funcs parse::email::*`  flag instead of allowing all functions with `--allow-funcs` without any arguments. This ensures that your SurrealDB instance will not be affected by vulnerabilities in the code of any other functions that a malicious actor could leverage to increase their privileges or execute unauthorized code.
```

Comment:
```suggestion
When running a SurrealDB server, you can configure the [capabilities](/docs/surrealdb/2.x/security/capabilities) for your SurrealQL queries. These capabilities are disabled by default to expose as little attack surface as possible to malicious actors. We strongly discourage running SurrealDB with all capabilities enabled without a good reason. Instead, we recommend enabling only the capabilities necessary for your service.

Whenever you need to enable a specific capability, we recommend doing it on a case-by-case basis instead of generally. For example, suppose you know that your queries need to be able to parse emails using functions. In that case, we recommend you run SurrealDB with the ` --allow-funcs parse::email::* ` flag instead of allowing all functions with` --allow-funcs` without arguments. 
Doing this ensures that your SurrealDB instance will not be affected by vulnerabilities in the code of any other functions that a malicious actor could leverage to increase their privileges or execute unauthorized code.
```